src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/helper/ScannedDocumentsHelper.java

```diff
@@ -44,6 +48,52 @@ private ScannedDocumentsHelper() {
             .collect(toList());
     }
 
+    public static void setExceptionRecordIdToScannedDocuments(
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseDetails
+    ) {
+        List<String> dcns = exceptionRecord.scannedDocuments == null
+            ? emptyList()
+            : exceptionRecord.scannedDocuments
+            .stream()
+            .map(scannedDocument -> scannedDocument.controlNumber)
+            .collect(toList());
+        @SuppressWarnings("unchecked")
+        Map<String, Object> caseData = (Map<String, Object>) caseDetails.caseData;
+        List<ScannedDocument> scannedDocuments = getScannedDocuments(caseData);
+        List<ScannedDocument> updatedScannedDocuments = scannedDocuments.stream()
+            .map(scannedDocument ->
+                dcns.contains(scannedDocument.controlNumber)
+                    ? new ScannedDocument(
+                    scannedDocument.fileName,
+                    scannedDocument.controlNumber,
+                    scannedDocument.type,
+                    scannedDocument.subtype,
+                    scannedDocument.scannedDate,
+                    scannedDocument.url,
+                    scannedDocument.deliveryDate,
+                    exceptionRecord.id
+                )
+                    : scannedDocument
+            )
+            .collect(toList());
+        caseData.put("scannedDocuments", updatedScannedDocuments);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static List<ScannedDocument> getScannedDocuments(Map<String, Object> caseData) {
```

Comment:
erm, that's same isn't it? just saw it few lines ago

Reply:
weeell, has extra features. but again querying `scannedDocuments` + null handling

Reply:
Used ternary operator. Further improvements in order to eliminate code duplication make code either a bit ugly or less performant (due to widely used streams).

Reply:
> due to widely used streams

yes... wrong programming language in use :shrug: